### DISCUSSION TITLE
Basic Powerwall Dashboard limited to 6 PW’s

### DISCUSSION THREAD

**Discussion by anderep** [jasonacox/Powerwall-Dashboard#253] [General]
I don’t know if this is a dashboard problem or something else, but I’m only seeing 6 powerwalls when browsing data for powerwall capacity and others. How can I show data for all powerwalls?

**Comment by jasonacox** [comment]
Hi @anderep - You are correct, the InfluxDB CQs and Dashboard panels are pre-programmed [up to 6 PWs](https://github.com/jasonacox/Powerwall-Dashboard#missing-string-data).  How many Powerwalls do you have?

**Reply by anderep** [reply]
Eight…

**Reply by jasonacox** [reply]
Thanks! I'll update the base install to accommodate at least 8 and have opened Issue https://github.com/jasonacox/Powerwall-Dashboard/issues/256 to track that.

As @ThePnuts mentions below, he created a custom solution to use InfluxDB 2 that supports this already, but we haven't figured out a clean way to merge that back into the push button setup/upgrade process. If you don't mind tinkering, look at his great work he contributed back.  Otherwise, I should have time this weekend to address the issue.

**Reply by jasonacox** [reply]
@anderep Can you paste the output from http://localhost:8675/freq (where localhost is the Dashboard host)?  I want to verify you are seeing PW1 ... PW8 data points.

**Reply by anderep** [reply]
yes, all data points

```
eric@opensprinkler:~/Powerwall-Dashboard/influxdb $ curl -s http://localhost:8675/freq
{"PW1_name": "TEPINV--1081100-23-V--CN322224D3V02K", "PW1_PINV_Fout": 60.028, "PW1_PINV_VSplit1": 125.4, "PW1_PINV_VSplit2": 125.5, "PW2_name": "TEPINV--1081100-23-V--CN322224D3V016", "PW2_PINV_Fout": 60.028, "PW2_PINV_VSplit1": 125.5, "PW2_PINV_VSplit2": 125.5, "PW3_name": "TEPINV--1081100-23-V--CN322224D3V02U", "PW3_PINV_Fout": 60.028, "PW3_PINV_VSplit1": 125.2, "PW3_PINV_VSplit2": 125.60000000000001, "PW4_name": "TEPINV--1081100-23-V--CN322248D3V08U", "PW4_PINV_Fout": 60.028, "PW4_PINV_VSplit1": 125.30000000000001, "PW4_PINV_VSplit2": 125.60000000000001, "PW5_name": "TEPINV--1081100-23-V--CN322263D3V042", "PW5_PINV_Fout": 60.027, "PW5_PINV_VSplit1": 125.30000000000001, "PW5_PINV_VSplit2": 125.60000000000001, "PW6_name": "TEPINV--1081100-23-V--CN322270D3V0BE", "PW6_PINV_Fout": 60.027, "PW6_PINV_VSplit1": 125.60000000000001, "PW6_PINV_VSplit2": 125.60000000000001, "PW7_name": "TEPINV--1081100-23-V--CN322277D3V01L", "PW7_PINV_Fout": 60.027, "PW7_PINV_VSplit1": 125.2, "PW7_PINV_VSplit2": 125.5, "PW8_name": "TEPINV--1081100-23-V--CN322277D3V00I", "PW8_PINV_Fout": 60.027, "PW8_PINV_VSplit1": 125.0, "PW8_PINV_VSplit2": 125.5, "ISLAND_VL1N_Main": 126.0, "ISLAND_FreqL1_Main": 60.03, "ISLAND_VL1N_Load": 126.0, "ISLAND_FreqL1_Load": 60.03, "ISLAND_PhaseL1_Main_Load": 0.0, "ISLAND_VL2N_Main": 126.0, "ISLAND_FreqL2_Main": 60.03, "ISLAND_VL2N_Load": 126.0, "ISLAND_FreqL2_Load": 60.03, "ISLAND_PhaseL2_Main_Load": 0.0, "ISLAND_VL3N_Main": 0.0, "ISLAND_FreqL3_Main": 0.0, "ISLAND_VL3N_Load": 0.0, "ISLAND_FreqL3_Load": 0.0, "ISLAND_PhaseL3_Main_Load": -180.5, "ISLAND_L1L2PhaseDelta": -180.5, "ISLAND_L1L3PhaseDelta": -180.5, "ISLAND_L2L3PhaseDelta": -180.5, "ISLAND_GridState": "ISLAND_GridState_Grid_Compliant", "ISLAND_L1MicrogridOk": true, "ISLAND_L2MicrogridOk": true, "ISLAND_L3MicrogridOk": false, "ISLAND_ReadyForSynchronization": true, "ISLAND_GridConnected": true, "METER_X_CTA_InstRealPower": -184.0, "METER_X_CTB_InstRealPower": 176.0, "METER_X_CTC_InstRealPower": 0.0, "METER_X_CTA_InstReactivePower": -1131.0, "METER_X_CTB_InstReactivePower": -1267.0, "METER_X_CTC_InstReactivePower": 0.0, "METER_X_LifetimeEnergyImport": 3358648.0, "METER_X_LifetimeEnergyExport": 7645914.0, "METER_X_VL1N": 125.36, "METER_X_VL2N": 125.24000000000001, "METER_X_VL3N": 0.0, "METER_X_CTA_I": 14.025500000000001, "METER_X_CTB_I": 13.573500000000001, "METER_X_CTC_I": 0.0, "METER_Y_CTA_InstRealPower": 0.0, "METER_Y_CTB_InstRealPower": 0.0, "METER_Y_CTC_InstRealPower": 0.0, "METER_Y_CTA_InstReactivePower": 0.0, "METER_Y_CTB_InstReactivePower": 0.0, "METER_Y_CTC_InstReactivePower": 0.0, "METER_Y_LifetimeEnergyImport": 2202460.0, "METER_Y_LifetimeEnergyExport": 2569.0, "METER_Y_VL1N": 125.36, "METER_Y_VL2N": 125.25, "METER_Y_VL3N": 0.0, "METER_Y_CTA_I": 0.0, "METER_Y_CTB_I": 0.0, "METER_Y_CTC_I": 0.0, "grid_status": 1}eric@opens
```

**Comment by ThePnuts** [comment]
It's not as straight forward as the default install, but if you want to follow along with what I have done using influxdb 2+, everything I updated to Flux supports many more then 8 powerwalls. If you are already comfortable with how to setup dockers, it would make this fairly easy. The entire setup lives in 5 dockers.

https://github.com/jasonacox/Powerwall-Dashboard/tree/main/tools/influxdb2

It also might not be hard to modify the original to support 2 more either, but I might not be the best help with that side of things.

**Comment by anderep** [comment]
Thank you @jasonacox 
Thanks @ThePnuts.  I don’t know docker, but I can copy paste and modify:P

**Reply by jasonacox** [reply]
I have added support for up to 12 Powerwalls in the base setup.  You can run the `./upgrade.sh` script to fetch the latest version and your Powerwalls should show up. You will need to import the new `dashboard.json` that contains the PW7 to PW12 additions.

If you try it, let us know if it works. 👍

**Reply by anderep** [reply]
When I upgrade, do I need to redo the customizations for the neurio data points?
Is there a way to import dashboard?  Last time I just cut and pasted.

**Reply by jasonacox** [reply]
The upgrade won't impact your `telegraf` or `influxdb` updates.  However, when you import the `dashboard.json` it will remove any edits you make.  Here is how to import the `dashboard.json` file directly:

<img width="198" alt="image" src="https://user-images.githubusercontent.com/836718/235581807-d19b1acc-3c45-4163-b8a2-85473ab39a44.png">

Click Import and then "Upload JSON File"

<img width="429" alt="image" src="https://user-images.githubusercontent.com/836718/235581851-33e32230-5426-4ac2-8c59-17d93050b8ea.png">

Select the file and "Open"

You will get this - select drop downs for "InfluxDB" and "Sun and Moon":

<img width="550" alt="image" src="https://user-images.githubusercontent.com/836718/235582024-4d694f8e-f98e-4de4-9185-6da4cd61ecd3.png">

Click "Import (Overwrite)"
